packages/libs/components/src/plots/VolcanoPlot.tsx

```diff
@@ -0,0 +1,274 @@
+   * Plot styles
+   * (can eventually be moved to a new file and applied as a visx theme)
+   */
+  const thresholdLineStyles = {
+    stroke: '#aaaaaa',
+    strokeWidth: 1,
+    strokeDasharray: 3,
+  };
+  const axisStyles = {
+    stroke: '#bbbbbb',
+    strokeWidth: 1,
+  };
+  const gridStyles = {
+    stroke: '#dddddd',
+    strokeWidth: 0.5,
+  };
+  console.log(yMin);
+
+  return (
+    // From docs " For correct tooltip positioning, it is important to wrap your
+    // component in an element (e.g., div) with relative positioning."
+    <div style={{ position: 'relative' }}>
+      <XYChart
+        height={300}
+        xScale={{ type: 'linear', domain: [xMin, xMax] }}
+        yScale={{ type: 'linear', domain: [yMin, yMax], zero: false }}
+        width={300}
+      >
+        <Grid numTicks={6} lineStyle={gridStyles} />
+        <Axis orientation="left" label="-log10 Raw P Value" {...axisStyles} />
+        <Axis orientation="bottom" label="log2 Fold Change" {...axisStyles} />
+
+        {/* Draw threshold lines as annotations below the data points */}
+        {significanceThreshold && (
+          <Annotation
+            datum={{
+              x: 0, // horizontal line so x could be anything
+              y: -Math.log10(Number(significanceThreshold)),
+            }}
+            {...thresholdLineAccessors}
+          >
+            <AnnotationLineSubject
+              orientation="horizontal"
+              {...thresholdLineStyles}
+            />
+          </Annotation>
+        )}
+        {log2FoldChangeThreshold && (
+          <>
+            <Annotation
+              datum={{
+                x: -log2FoldChangeThreshold,
+                y: 0, // vertical line so y could be anything
+              }}
+              {...thresholdLineAccessors}
+            >
+              <AnnotationLineSubject {...thresholdLineStyles} />
+            </Annotation>
+            <Annotation
+              datum={{
+                x: log2FoldChangeThreshold,
+                y: 0, // vertical line so y could be anything
+              }}
+              {...thresholdLineAccessors}
+            >
+              <AnnotationLineSubject {...thresholdLineStyles} />
+            </Annotation>
+          </>
+        )}
+
+        {/* The data itself */}
+        <Group opacity={markerBodyOpacity ?? 1}>
+          <GlyphSeries
+            dataKey={'data'}
+            data={dataPoints}
+            {...dataAccessors}
+            colorAccessor={(d) => {
+              return d.color;
+            }}
+          />
+        </Group>
+      </XYChart>
+    </div>
+  );
+}
+
+/**
+ * Assign color to point based on significance and magnitude change thresholds
+ */
+function assignSignificanceColor(
+  xValue: number, // has already been log2 transformed
+  yValue: number, // the raw pvalue
+  significanceThreshold: number,
+  log2FoldChangeThreshold: number,
+  significanceColors: string[] // Assuming the order is [high, low, not significant]
```

Comment:
**suggestion**: This isn't at all unclear code or anything complicated, so this is really just a suggestion. I don't know if it makes the code better. Given this assumption, you could name the indicies:

```js
const HIGH = 0;
const LOW = 1;
const INSIGNIFICANT = 2;

 if (yValue >= significanceThreshold) {
    return significanceColors[INSIGNIFICANT];
  }

 if (xValue >= log2FoldChangeThreshold) {
    return significanceColors[HIGH];
  }
...
```